tsfresh/feature_extraction/extraction.py

```diff
@@ -103,9 +114,103 @@ def extract_features(timeseries_container, feature_extraction_settings=None,
         profiling.end_profiling(profiler, filename=feature_extraction_settings.PROFILING_FILENAME,
                                 sorting=feature_extraction_settings.PROFILING_SORTING)
 
+    return result
+
+
+def _extract_features_parallel_per_kind(kind_to_df_map, settings, column_id, column_value):
+    """
+    Parallelize the feature extraction per kind.
+
+    :param timeseries_container: The pandas.DataFrame with the time series to compute the features for, or a
+               dictionary of pandas.DataFrames.
+    :type timeseries_container: pandas.DataFrame or dict
+
+    :param column_id: The name of the id column to group by.
+    :type column_id: str
+    :param column_value: The name for the column keeping the value itself.
+    :type column_value: str
+
+    :param settings: settings object that controls which features are calculated
+    :type settings: tsfresh.feature_extraction.settings.FeatureExtractionSettings
+
+    :return: The (maybe imputed) DataFrame containing extracted features.
+    :rtype: pandas.DataFrame
+    """
+    all_possible_unique_id_values = set(id_value for kind, df in kind_to_df_map.items()
```

Comment:
I left this part untouched and just copied it from the pevious function. From what I observed, there is not much speed to gain here. We should settle on the most readable solution.

Besides, I'm not entirely sure we actually need this part. It is there to ensure that the feature matrix contains entries for all samples. Shouldn't the feature calculators return at least NaN in any case?